pkg/resource/function/hooks.go

```diff
@@ -400,60 +392,30 @@ func (rm *resourceManager) updateFunctionArchitectures(
 		input.Architectures = nil
 	}
 
-	if latest.ko.Spec.Code != nil {
-		if latest.ko.Spec.PackageType != nil && *latest.ko.Spec.PackageType == "Image" {
-			input.ImageUri = latest.ko.Spec.Code.ImageURI
-		} else if latest.ko.Spec.PackageType != nil && *latest.ko.Spec.PackageType == "Zip" {
-			input.S3Bucket = latest.ko.Spec.Code.S3Bucket
-			input.S3Key = latest.ko.Spec.Code.S3Key
-		}
-	}
-
-	_, err = rm.sdkapi.UpdateFunctionCodeWithContext(ctx, input)
-	rm.metrics.RecordAPICall("UPDATE", "UpdateFunctionArchitectures", err)
-	if err != nil {
-		return err
-	}
-
-	return nil
-}
-
-// updateFunctionsCode calls UpdateFunctionCode to update a specific lambda
-// function code.
-func (rm *resourceManager) updateFunctionCode(
-	ctx context.Context,
-	desired *resource,
-	delta *ackcompare.Delta,
-) error {
-	var err error
-	rlog := ackrtlog.FromContext(ctx)
-	exit := rlog.Trace("rm.updateFunctionCode")
-	defer exit(err)
-
-	if delta.DifferentAt("Spec.Code.S3Key") &&
-		!delta.DifferentAt("Spec.Code.S3Bucket") &&
-		!delta.DifferentAt("Spec.Code.S3ObjectVersion") &&
-		!delta.DifferentAt("Spec.Code.ImageURI") {
-		log := ackrtlog.FromContext(ctx)
-		log.Info("updating code.s3Key field is not currently supported.")
-		return nil
-	}
-
-	dspec := desired.ko.Spec
-	input := &svcsdk.UpdateFunctionCodeInput{
-		FunctionName: aws.String(*dspec.Name),
-	}
-
 	if dspec.Code != nil {
-		switch {
-		case dspec.Code.ImageURI != nil:
-			input.ImageUri = dspec.Code.ImageURI
-		case dspec.Code.S3Bucket != nil,
-			dspec.Code.S3Key != nil,
-			dspec.Code.S3ObjectVersion != nil:
-			input.S3Bucket = dspec.Code.S3Bucket
-			input.S3Key = dspec.Code.S3Key
-			input.S3ObjectVersion = dspec.Code.S3ObjectVersion
+		if delta.DifferentAt("Spec.Code.SHA256") && dspec.Code.SHA256 != nil {
+			if dspec.Code.S3Key != nil {
+				input.S3Key = aws.String(*dspec.Code.S3Key)
+			}
+			if dspec.Code.S3Bucket != nil {
+				input.S3Bucket = aws.String(*dspec.Code.S3Bucket)
+			}
+			if dspec.Code.S3ObjectVersion != nil {
+				input.S3ObjectVersion = aws.String(*dspec.Code.S3ObjectVersion)
+			}
+		} else if delta.DifferentAt("Spec.Code.ImageURI") && dspec.Code.ImageURI != nil {
+			if dspec.Code.ImageURI != nil {
+				input.ImageUri = aws.String(*dspec.Code.ImageURI)
+			}
+
+		} else { // We need to pass the latest code to Update API call,
+			//if there is change in architecture and no change in Code
+			if latest.ko.Spec.PackageType != nil && *latest.ko.Spec.PackageType == "Image" {
+				input.ImageUri = latest.ko.Spec.Code.ImageURI
```

Comment:
I think we only need to update the image if the ImageURI changed

Reply:
This code is for passing `code` values if there is change in `Architecture` field. To update `Architecture` field we need to pass the code again too, this handles case that even if there is no change in code, it will just pass the current values of it along with updated architecture field.